tensorflow/stream_executor/BUILD

```diff
@@ -511,6 +515,7 @@ cc_library(
         "stream_executor_pimpl.cc",
     ],
     hdrs = ["stream_executor_pimpl.h"],
+    copts = if_cuda_is_configured(["-DGOOGLE_CUDA=1"]),
```

Comment:
I'm looking at https://github.com/tensorflow/tensorflow/blob/1acfde6e42c1007b028c82833c5922f14023d6fe/tensorflow/stream_executor/stream_executor_pimpl.cc#L303 : is it possible to remove the `#if`?

Reply:
That means the cuda_dnn.h would be also included for the non-GPU env (Is that OK?), because `gpu::CudnnSupport` is defined in it and included here:
https://github.com/tensorflow/tensorflow/blob/1acfde6e42c1007b028c82833c5922f14023d6fe/tensorflow/stream_executor/stream.h#L46

Reply:
Or, how about defining a dummy CudnnSupport for the CPU in the stream_executor_pimpl.cc, like:
```c++
#if !GOOGLE_CUDA
class CudnnSupport : public dnn::DnnSupport {
...
  Status GetFusedConvolveExecutionPlans(...) {
  }
}
#endif
```

Reply:
Nah that's too much work. Let's go with your approach, but instead of defining GOOGLE_CUDA, replace `cc_library` with `tf_cuda_library`.